core/src/main/java/io/questdb/std/Unsafe.java

```diff
@@ -213,25 +220,37 @@ public static sun.misc.Unsafe getUnsafe() {
 
     public static long malloc(long size, int memoryTag) {
         try {
```

Comment:
It's unlikely but this approach could spread OOMs from current thread into other mallocating threads on a big malloc that's going to fail . 
On the other hand using RAM.getAndUpdate() to update var conditionally would probably be too costly .

Reply:
That's good point. I don't know how to solve this tbh without costly CAS loop so I will leave it is